hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -756,6 +814,26 @@ public boolean isGrpcBlockTokenEnabled() {
     return grpcBlockTokenEnabled;
   }
 
+  /**
+   * Return config value of {@link OMConfigKeys#OZONE_OM_MULTITENANCY_ENABLED}.
+   */
+  public boolean isS3MultiTenancyEnabled() {
+    return isS3MultiTenancyEnabled;
+  }
+
+  /**
+   * Throws OMException if S3 multi-tenancy is not enabled.
+   */
+  public void checkS3MultiTenancyEnabled() throws OMException {
+    if (isS3MultiTenancyEnabled) {
+      return;
+    }
+
```

Comment:
The `FEATURE_NOT_ENABLED` OMException should be logged somewhere, perhaps a LOG.debug?
This exception looks to propagate up and return to the client in the `OMResponse` with not Success, exception msg and status` FEATURE_NOT_ENABLED`.  The `OzoneManagerProtocolClientSideTranslatorPB` wraps the error in an `IOException` for the client.  In the case of the s3gateway this most likely results in an ambiguous `500 http response` with `INTERNAL_ERROR`.  It would be great to capture the `FEATURE_NOT_ENABLED` error in a `debug` log.

Reply:
Hi @neils-dev , thanks for the comment.

I understand the concern. Though the current logic in `OzoneManager#getS3VolumeContext` will not actually reach the multi-tenancy branch. This means all attempts to access a tenant volume with a previously generated tenant accessId will be redirected to the default s3v if the feature is disabled.

If we do need the check at some point we can easily add the logic at wherever getVolume() is called in S3G. For example here: https://github.com/apache/ozone/blob/3fd7dc6c77d2eb9a43635c91aa7654e0bdcdc72a/hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/BucketEndpoint.java#L494-L497

Reply:
And thanks for pointing this out! I have made the logic a bit cleaner in `getS3VolumeContext()`: 02ee31ee285a521716c4437dffe7415a61b07937